### PR TITLE
Support auto height in SimpleSheet

### PR DESCRIPTION
**Pull request by benisgold**
Fixes APP-####

## What changed (plus any additional context for devs)
* if you do `<SimpleSheet height="auto" />`, the height of the sheet will conform to the height of its children.

## Screen recordings / screenshots

https://user-images.githubusercontent.com/15272675/229940798-9420cc79-52c1-4ba2-8b77-dd2fcf466862.mov

https://user-images.githubusercontent.com/15272675/229940849-f6c6a800-f394-4a29-aada-c224c23cb949.mp4

## What to test
hardware wallet sheets use SimpleSheet so might want to double check that

### CODE REVIEW

File: src/navigation/Routes.android.js [hunk 1]
```diff
@@ -323,7 +323,7 @@ function BSNavigator() {
         options={({ route: { params = {} } }) => {
           return {
             ...bottomSheetPreset,
-            height: params.sheetHeight,
+            height: params.sheetHeight ?? 0,
```
Comment (benisgold) [inline src/navigation/Routes.android.js:326]: don't want to show the sheet until we set the height

File: src/navigation/config.js [hunk 2]
```diff
@@ -369,7 +369,7 @@ export const explainSheetV2Config = {
   options: ({ route: { params = {} } }) => ({
     ...buildCoolModalConfig({
       ...params,
-      longFormHeight: params.sheetHeight,
+      longFormHeight: params.sheetHeight ?? -safeAreaInsetValues.bottom,
```
Comment (benisgold) [inline src/navigation/config.js:372]: don't want to show the sheet until we set the height. on ios, the bottom margin of the cool modal is above the bottom safe area inset (unlike android, where it's actually the true screen bottom)

File: src/components/sheet/SimpleSheet.tsx [hunk 3]
```diff
@@ -43,10 +56,12 @@ export const SimpleSheet = ({
           backgroundColor: backgroundColor,
         }}
         contentContainerStyle={{
-          minHeight: customHeight ?? fullSheetHeight,
+          minHeight: sheetHeight,
         }}
       >
-        {children}
+        <View onLayout={e => setAutoHeight(e.nativeEvent.layout.height)}>
```
Comment (skylarbarrera) [inline src/components/sheet/SimpleSheet.tsx:62]: king

File: src/navigation/config.js (hunk 2 above)
Comment (estrattonbailey) [reply src/navigation/config.js:372]: You can get rid of `params.sheetHeight` in both places now too 👍

**Review by estrattonbailey — APPROVED** [review]
NICE

**Comment by jkadamczyk** [comment]
THAT"S GREAT!

**Pull request closed without merging**